src/Middleware/HttpLogging/src/HttpLoggingServicesExtensions.cs

```diff
@@ -15,22 +17,37 @@ public static class HttpLoggingServicesExtensions
     /// </summary>
     /// <param name="services">The <see cref="IServiceCollection"/> for adding services.</param>
     /// <param name="configureOptions">A delegate to configure the <see cref="HttpLoggingOptions"/>.</param>
-    /// <returns></returns>
+    /// <returns>The original service collection for chaining.</returns>
     public static IServiceCollection AddHttpLogging(this IServiceCollection services, Action<HttpLoggingOptions> configureOptions)
```

Comment:
Is it worth registering `HttpLoggingMiddleware` as a singleton here?

Reply:
But it should implement `IMiddleware` in that case: https://source.dot.net/#Microsoft.AspNetCore.Http.Abstractions/Extensions/UseMiddlewareExtensions.cs,48

Reply:
No, the IMiddleware approach is less efficient.